=== modified file 'doc/tutorial/exercises/03/control-migrations/keyvalueserver.c'
--- doc/tutorial/exercises/03/control-migrations/keyvalueserver.c	2012-08-20 20:46:17 +0000
+++ doc/tutorial/exercises/03/control-migrations/keyvalueserver.c	2012-08-28 17:42:43 +0000
@@ -83,16 +83,20 @@
 }
 
 int
-main(int argc, char **argv)
+main(int argc, char **argv) NOTE_LOCALS
 {
   int server_sock = 0, client_sock = 0, 
     client_addrlen = sizeof(struct sockaddr_in);
   struct sockaddr_in client_addr;
 
-  init_store();
-  server_sock = init_listen_socket(5000);
+  if (!kitsune_is_updating()) {
+    init_store();
+    server_sock = init_listen_socket(5000);
+  }
   assert(server_sock > 0);
-  
+
+  if (!kitsune_is_updating_from("client"))
+    handle_client(client_sock);
   while (1) {
     kitsune_update("main");
     client_sock = accept(server_sock, &client_addr, 

